### PR TITLE
enh(dashboards) :  display a warning message to save before quit

### PR DESCRIPTION
**Pull request by Med1233**
## Description

display a warning message to save before quit

**Fixes** # [(issue)](https://centreon.atlassian.net/browse/MON-23241)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by jeremyjaouen** [comment]
LGTM but we discuss with @Thebarda to use the PageHeaderActions component instead of a new one. Your solution is good to me anyway.

File: centreon/packages/ui/src/components/Header/PageHeader/PageHeader.styles.ts [hunk 1]
```diff
@@ -53,6 +56,38 @@ const useStyles = makeStyles()((theme) => ({
...
+      letterSpacing: '0.15px'
+    }
+  },
+  pageHeaderMessageIcon: {
+    alignItems: 'center',
+    background: theme.palette.warning.main,
+    borderRadius: '50%',
+    color: theme.palette.common.white,
+    display: 'flex',
+    font: 'normal normal 500 20px/20px Roboto',
```
Comment (Thebarda) [inline centreon/packages/ui/src/components/Header/PageHeader/PageHeader.styles.ts:75]: Avoir custom font.
Use the ones from mui theme

File: centreon/packages/ui/src/components/Header/PageHeader/PageHeader.styles.ts [hunk 2]
```diff
@@ -53,6 +56,38 @@ const useStyles = makeStyles()((theme) => ({
     position: 'relative',
     transform: `translateY(-${theme.spacing(0.25)})`
   },
+  pageHeaderMessage: {
+    alignItems: 'center',
+    color: theme.palette.warning.main,
+    display: 'flex',
+    gap: theme.spacing(1),
+    h2: {
+      font: 'normal normal 500 20px/20px Roboto',
```
Comment (Thebarda) [inline centreon/packages/ui/src/components/Header/PageHeader/PageHeader.styles.ts:65]: Same comment about the font

File: centreon/packages/ui/src/components/Header/PageHeader/PageHeaderMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const { classes } = useStyles();
+
+  return (
+    <Box className={classes.pageHeaderMessage}>
+      {message && (
+        <>
+          <Box className={classes.pageHeaderMessageIconWrapper}>
+            <Box className={classes.pageHeaderMessageIcon}>!</Box>
+          </Box>
+          <Typography aria-label="unsaved changes warning" variant="h2">
```
Comment (Thebarda) [inline centreon/packages/ui/src/components/Header/PageHeader/PageHeaderMessage.tsx:23]: The aria label shoud be changed

**Comment by Med1233** [comment]
> LGTM but we discuss with @Thebarda to use the PageHeaderActions component instead of a new one. Your solution is good to me anyway.

I think this is an exception because we need a different style for this new component

**Comment by Thebarda** [comment]
Is it done?

**Comment by Med1233** [comment]
> Is it done?

yes

**Review by Noha-ElAbrouki — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**